libs/dal/src/repositories/topic/topic.repository.ts

```diff
@@ -0,0 +1,25 @@
+import { AuthProviderEnum } from '@novu/shared';
+import { FilterQuery } from 'mongoose';
+
+import { EnvironmentId, OrganizationId, TopicKey, TopicEntity } from './topic.entity';
+import { Topic } from './topic.schema';
+
+import { BaseRepository } from '../base-repository';
+
+export class TopicRepository extends BaseRepository<FilterQuery<TopicEntity>, TopicEntity> {
```

Comment:
@djabarovgeorge, how should I enforce envId and orgId to be mandatory? Seeing references of the changes you did I have not clear if I already enforced it here.

Reply:
I think that this part will do the job.

```suggestion

class PartialTopicEntity extends Omit(TopicEntity, ['_environmentId', '_organizationId']) {}

type EnforceEnvironmentQuery = FilterQuery<PartialTopicEntity & Document> &
  ({ _environmentId: string } | { _organizationId: string });

export class TopicRepository extends BaseRepository<EnforceEnvironmentQuery, TopicEntity> {
```